README.md

```diff
@@ -1,15 +1,195 @@
 
-TBD: Intro
+This Terraform module allows you to provision an AWS GraphDB cluster within a Virtual Private Cloud (VPC) in AWS. The module provides a flexible way to configure the cluster and the associated VPC components.
 
-## How to use this module
+—
+## Prerequisites
+
+Before you begin using this Terraform module, ensure you have the following prerequisites:
+
+- **Terraform Installed**: You should have Terraform installed on your local machine. You can download Terraform from the [official website](https://www.terraform.io/downloads.html).
+
+- **AWS Credentials**: Ensure that you have AWS credentials configured on your machine. You can configure AWS access keys and secret keys using the [AWS CLI](https://aws.amazon.com/cli/).
+
+- **Terraform AWS Provider Configuration**: Configure the AWS provider in your Terraform project. You can add your AWS access and secret keys as environment variables or use other methods for provider configuration.
+
+- **Terraform Configuration File**: Create a new Terraform configuration file in your project, and use this module by specifying its source.
+
+—
+
+## Usage
+
+To use this module, follow these steps:
+
+1. Clone the repository to your local machine:
+
+   ```shell
+   git clone https://github.com/Ontotext-AD/terraform-aws-graphdb.git
+   ```
+
+2. Navigate to the module directory:
+
+   ```shell
+   cd terraform-aws-graphdb/modules/aws-graphdb-cluster
+   ```
+
+3. Create a new Terraform configuration in your project, and use this module by specifying its source:
+
+   ```hcl
+   module "graphdb_cluster" {
+     source = "../../modules/aws-graphdb-cluster"
+
+     # Define your configuration variables here, such as cluster size, VPC settings, and other parameters.
+
+
+   }
+
+   # Configure the required variables for the module.
+   ```
+
+
+
+
+4. Initialize your Terraform workspace and apply the changes:
+
+  ```shell
+    terraform init
+    terraform plan
+  ```
+
+5. Review and confirm the changes when prompted and apply if everything is okay.
+
+  ```shell
+    terraform apply
+  ```
+
+
+
+
+
+## Configuration
+
+### Important Variables (Inputs)
+
+The following are the important variables you should configure when using this module:
+
+- `vpc_id`: The ID of the Virtual Private Cloud (VPC) where you want to create the AWS GraphDB cluster. This is a required variable.
+
+- `instance_type`: The instance type for the GDB cluster nodes. This should match your performance and cost requirements.
+
+- `cluster_size`: The number of instances in the cluster. Recommended is 3, 5 or 7 in order to have consensus according to the [RAFT algorithm](https://raft.github.io/). 
+
+- `subnet_ids`: A list of subnet IDs in your VPC to place the Neptune instances. Ensure that these subnets have proper route tables and security groups configured.
+
+- `security_group_ids`: A list of security group IDs to associate with the instances for network security.
+
+TODO MORE
+
+### Example Configuration
```

Comment:
Could be moved as an example under examples/ or simply refer to the one that we already have.